llvm/lib/Target/AIE/AIEHazardRecognizer.cpp

```diff
@@ -441,6 +461,12 @@ ScheduleHazardRecognizer::HazardType AIEHazardRecognizer::getHazardType(
     iterator_range<const MachineOperand *> MIOperands,
     const MachineRegisterInfo &MRI, int DeltaCycles) const {
   const unsigned SchedClass = TII->getSchedClass(Desc, MIOperands, MRI);
+  if (!IsPreRA && SchedClass == 0 &&
```

Comment:
This verification could be refactored out to a helper function like `check...` or `validate...`, that can be used also below (same code).